dali/kernels/common/join/tensor_join_gpu_impl.cu

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/common/join/tensor_join_gpu_impl.h"
+#include "dali/kernels/common/join/tensor_join_gpu_impl.cuh"
+
+namespace dali {
+namespace kernels {
+namespace tensor_join {
+
+
+template <typename T, bool new_axis>
+KernelRequirements TensorJoinImplGPU<T, new_axis>::Setup(
+        KernelContext &ctx,
+        const std::function<const TensorListShape<> *(int)> &get_input_shape,
```

Comment:
`TensorListShape` is _heavy_. It contains a vector. Copying it would result in multiple dynamic allocations - and it's very unlikely to find a contiguous array of TensorListShape (or pointers to them) in the wild.